plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineProjectIdValidator.java

```diff
@@ -0,0 +1,29 @@
+package com.google.cloud.tools.eclipse.appengine.newproject;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
```

Comment:
Oh gosh, how does this relate to the Project ID shown on the developer console?  The [Go binding tests](https://github.com/golang/appengine/blob/master/internal/app_id_test.go) have other syntaxes too.